sentry/src/main/java/io/sentry/HostnameCache.java

```diff
@@ -21,12 +21,18 @@
  * by keeping track of the hostname for a period defined during the construction. For performance
  * purposes, the operation of retrieving the hostname will automatically fail after a period of time
  * defined by {@link #GET_HOSTNAME_TIMEOUT} without result.
+ *
+ * <p>HostnameCache is a singleton and its instance should be obtained through {@link
+ * HostnameCache#getInstance()}.
  */
 final class HostnameCache {
   private static final long HOSTNAME_CACHE_DURATION = TimeUnit.HOURS.toMillis(5);
 
   /** Time before the get hostname operation times out (in ms). */
   private static final long GET_HOSTNAME_TIMEOUT = TimeUnit.SECONDS.toMillis(1);
+
+  @Nullable private static HostnameCache INSTANCE;
```

Comment:
```suggestion
  @NotNull private static final HostnameCache INSTANCE = new HostnameCache();
```

Reply:
This will create `HostnameCache` instance when class is loaded, even if the actual instance has never been requested. As a result there can be nobody responsible for closing the thread pool.

Reply:
That makes sense, thanks.